### PR TITLE
[DFE-804] updates to MergeMetrics in BroadInternalRNAWithUMIs

### PR DESCRIPTION
**Pull request by mmorgantaylor**
Changes:
- convert a "?" metric to "NaN"
- round certain median metrics to integers (to support TDR integrations with GP - see [DFE-804](https://broadworkbench.atlassian.net/browse/DFE-804) for context)

Functional testing with TDR in Terra here: https://app.terra.bio/#workspaces/broad-dsp-spec-ops-fc/Twist%20TCAP%20Data%20Processing%20DEV/job_history/968f5796-313b-4729-ae19-c2c93307485f
specifically this sample: https://job-manager.dsde-prod.broadinstitute.org/jobs/bf250bb2-4fb7-4334-bf39-976e13245336
(many samples in that submission are failing because of missing file issues, not relevant to this PR)

### CODE REVIEW

File: pipelines/broad/internal/rna_seq/BroadInternalRNAWithUMIs.changelog.md [hunk 1]
```diff
@@ -1,3 +1,8 @@
+# 1.0.20
+2022-12-07 (Date of Last Commit)
+
+* In MergeMetrics task, convert "?" to "NaN" and round some median metrics to integers.
```
Comment (nikellepetrillo) [inline pipelines/broad/internal/rna_seq/BroadInternalRNAWithUMIs.changelog.md:4]:
```suggestion
* In MergeMetrics task, convert \"?\" to \"NaN\" and round some median metrics to integers.
```

Comment (nikellepetrillo) [reply pipelines/broad/internal/rna_seq/BroadInternalRNAWithUMIs.changelog.md:4]: We have to escape double quotes in our changelogs or else the jenkins release fails (i think it has a hard time parsing the quotes)

**Comment by nikellepetrillo** [comment]
retest this please

**Comment by nikellepetrillo** [comment]
test this please

**Comment by nikellepetrillo** [comment]
add to whitelist

**Comment by khajoue2** [comment]
test this please

File: tasks/broad/RNAWithUMIsTasks.wdl [hunk 2]
```diff
@@ -462,7 +462,12 @@ task MergeMetrics {
 
     for col in range(0, len(rows[0])):
       key = rows[0][col].lower()
-      print(f"{key}\t{rows[1][col]}")
+      value = rows[1][col]
+      if value == "?":
+        value = "NaN"
+      if key in ["median_insert_size", "median_absolute_deviation", "median_read_length", "hq_median_mismatches"]:
+        value = str(int(float(value)))
```
Comment (kachulis) [inline tasks/broad/RNAWithUMIsTasks.wdl:469]: don't need to cast to a `str`, the f-string will take care of this for you.

**Review by kachulis — APPROVED** [review]

File: tasks/broad/RNAWithUMIsTasks.wdl (hunk 2 above)
Comment (mmorgantaylor) [reply tasks/broad/RNAWithUMIsTasks.wdl:469]: yeah, I went back and forth on this - I left it with the str() conversion just to return `value` to the same type as it started as. if you have strong feelings about this, happy to change it.

**Review by khajoue2 — APPROVED** [review]
lgtm

**Review by ekiernan — APPROVED** [review]

**Comment by timaeusx** [comment]
Closing and reopening to refresh PR checks

**Pull request closed without merging**